File: backend/app/v1/schemas/etl.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class ETLRunResponse(BaseModel):
    audit_id: int
    status: str
    started_at: datetime
    finished_at: Optional[datetime] = None
    duration_ms: Optional[int] = None
    history_new: int
    history_skipped: int
    artists_new: int
    tracks_new: int
    cursor_next_ms: Optional[int] = None
    error_message: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class ETLStatusResponse(BaseModel):
    runs: list[ETLRunResponse]
    total: int

File: frontend/src/types/etl.ts
export interface ETLRun {
  audit_id: number;
  status: string;
  started_at: string;
  finished_at: string | null;
  duration_ms: number | null;
  history_new: number;
  history_skipped: number;
  artists_new: number;
  tracks_new: number;
  cursor_next_ms: number | null;
  error_message: string | null;
}

export interface ETLStatusResponse {
  runs: ETLRun[];
  total: number;
}

export interface ETLRunResponse {
  audit_id: number;
  status: string;
  started_at: string;
  finished_at: string | null;
  duration_ms: number | null;
  history_new: number;
  history_skipped: number;
  artists_new: number;
  tracks_new: number;
  cursor_next_ms: number | null;
  error_message: string | null;
}
